app/models/unit.rb

```diff
@@ -1,3 +1,17 @@
 class Unit < ActiveRecord::Base
   has_one :renter
+  belongs_to :owner
+
+  validates :name, :owner_id, :sq_footage, :rent, :status, presence: true
+
+  enum status: ["not_rented", "rented"]
+
+  def self.avg_cost_per_sq_ft
+    @units = Unit.all
+    @units.average(:cost_per_sq_ft)
+  end
+
+  def cost_per_sq_ft
```

Comment:
Be careful with this: believe you have an attribute in your database and an instance method on your model named the same thing, so when you call `@unit.cost_per_sq_ft` which are you calling?

This is possibly *slightly* more complicated than was intended, but you have a couple of options when working with cost per square foot: 1) save it on the database (in which case I would recommend calculating it *for* your user instead of letting them calculate it themselves, which would require a `before_save` callback), or 2) not saving it on the database and always using a calculation in ActiveRecord when you want that data (results in a more complicated `.avg_cost_per_sq_ft` calculation).

I think this is fine for now, but something to keep in mind! Will be important to weigh those benefits in later work. Let me know if you'd like to discuss!